src/value.jl

```diff
@@ -1,145 +1,181 @@
-    if x.pool !== y.pool
+function Base.:<(x::CatValue, y::CatValue)
+    if pool(x) !== pool(y)
         throw(ArgumentError("CategoricalValue objects with different pools cannot be tested for order"))
-    elseif !isordered(x.pool) # !isordered(y.pool) is implied by x.pool === y.pool
+    elseif !isordered(pool(x)) # !isordered(pool(y)) is implied by pool(x) === pool(y)
         throw(ArgumentError("Unordered CategoricalValue objects cannot be tested for order using <. Use isless instead, or call the ordered! function on the parent array to change this"))
     else
-        return order(x.pool)[x.level] < order(y.pool)[y.level]
+        return order(x) < order(y)
     end
 end
 
-Base.get(x::CategoricalValue{T,R}) where {T,R} = convert(T,x)
-
-# AbstractString interface
-Base.string(x::CategoricalValue{<:AbstractString}) = get(x)
-Base.length(x::CategoricalValue{<:AbstractString}) = length(get(x))
-Base.endof(x::CategoricalValue{<:AbstractString}) = endof(get(x))
-Base.sizeof(x::CategoricalValue{<:AbstractString}) = sizeof(get(x))
-Base.nextind(x::CategoricalValue{<:AbstractString}, i::Integer) = nextind(get(x), i)
-Base.prevind(x::CategoricalValue{<:AbstractString}, i::Integer) = prevind(get(x), i)
-Base.next(x::CategoricalValue{<:AbstractString}, i::Int) = next(get(x), i)
-Base.done(x::CategoricalValue{<:AbstractString}, i::Int) = done(get(x), i)
-Base.getindex(x::CategoricalValue{<:AbstractString}, i::Int) = getindex(get(x), i)
-Base.codeunit(x::CategoricalValue{<:AbstractString}, i::Integer) = codeunit(get(x), i)
-Base.ascii(x::CategoricalValue{<:AbstractString}) = ascii(get(x))
-Base.isvalid(x::CategoricalValue{<:AbstractString}) = isvalid(get(x))
-Base.isvalid(x::CategoricalValue{<:AbstractString}, i::Integer) = isvalid(get(x), i)
-Base.match(r::Regex, s::CategoricalValue{<:AbstractString},
+# AbstractString interface for CategoricalString
+Base.string(x::CategoricalString) = get(x)
+Base.length(x::CategoricalString) = length(get(x))
+Base.endof(x::CategoricalString) = endof(get(x))
+Base.sizeof(x::CategoricalString) = sizeof(get(x))
+Base.nextind(x::CategoricalString, i::Integer) = nextind(get(x), i)
+Base.prevind(x::CategoricalString, i::Integer) = prevind(get(x), i)
+Base.next(x::CategoricalString, i::Int) = next(get(x), i)
+Base.getindex(x::CategoricalString, i::Int) = getindex(get(x), i)
+Base.codeunit(x::CategoricalString, i::Integer) = codeunit(get(x), i)
+Base.ascii(x::CategoricalString) = ascii(get(x))
+Base.isvalid(x::CategoricalString) = isvalid(get(x))
+Base.isvalid(x::CategoricalString, i::Integer) = isvalid(get(x), i)
+Base.match(r::Regex, s::CategoricalString,
            idx::Integer=start(s), add_opts::UInt32=UInt32(0)) =
     match(r, get(s), idx, add_opts)
-Base.matchall(r::Regex, s::CategoricalValue{<:AbstractString}, overlap::Bool=false) =
+Base.matchall(r::Regex, s::CategoricalString, overlap::Bool=false) =
     matchall(r, get(s), overlap)
+Base.collect(x::CategoricalString) = collect(get(x))
```

Comment:
Don't forget to test this (or leave for another PR if that's not really required for this one).

Reply:
Done.